api/v1alpha1/clienttrafficpolicy_types.go

```diff
@@ -84,6 +88,73 @@ type ClientTrafficPolicySpec struct {
 	HTTP1 *HTTP1Settings `json:"http1,omitempty"`
 }
 
+// ClientIPDetectionSettings provides XFF and extension configuration for client IP detection on the listener.
+//
+// +kubebuilder:validation:XValidation:rule="!(has(self.xffNumTrustedHops) && has(self.extensions))",message="extensions cannot be used in conjunction with xffNumTrustedHops"
+type ClientIPDetectionSettings struct {
```

Comment:
prefer if we make incremental changes, and this PR includes only the ability to set `numTrustedHops` to be able to retrieve the correct client IP from XFF.
```
clientIPDetection:
  xForwardedFor:
    numTrustedHops: <>
```